consensus/consensus_service.go

```diff
@@ -603,3 +601,31 @@ func (consensus *Consensus) SetBlockNum(blockNum uint64) {
 	defer consensus.mutex.Unlock()
 	consensus.blockNum = blockNum
 }
+
+// read the payload for signature and bitmap; offset is the beginning position of reading
+func (consensus *Consensus) readSignatureBitmapPayload(recvPayload []byte, offset int) (*bls.Sign, *bls_cosi.Mask, error) {
+	if offset+48 > len(recvPayload) {
+		return nil, nil, errors.New("payload not have enough length")
+	}
+	payload := append(recvPayload[:0:0], recvPayload...)
+	//#### Read payload data
+	// 48 byte of multi-sig
+	multiSig := payload[offset : offset+48]
+	offset += 48
+	// bitmap
+	bitmap := payload[offset:]
```

Comment:
check the length of payload, if it is less than offset, you will have panic.

Reply:
the bitmap payload is variable length, so I can only make sure the length is greater than 0

Reply:
checked in the beginning

Reply:
the first if check should be fine.